buildSrc/src/main/kotlin/Train.kt

```diff
@@ -46,35 +47,26 @@ fun Project.filterSnapshotTests() {
 }
 
 fun Project.setupTrainForSubproject() {
-    val build_snapshot_train: String? by extra
-    if (!build_snapshot_train.toBoolean()) {
-        return
-    }
+    if (!buildSnapshotTrain) return
 
-    val atomicfu_version: String by extra
-    val coroutines_version: String by extra
-    val serialization_version: String by extra
+    val kotlinVersion = rootProject.libs.versions.kotlin.get()
+    println("Using Kotlin $kotlinVersion for project $this")
 
-    extra["kotlin_version"] = rootProject.properties["kotlin_snapshot_version"]
-    val kotlin_version: String by extra
-    println("Using Kotlin $kotlin_version for project $this")
     val deployVersion = properties["DeployVersion"]
     if (deployVersion != null) version = deployVersion
 
-    val skipSnapshotChecks = rootProject.properties["skip_snapshot_checks"] != null
-    if (!skipSnapshotChecks) {
-        check(version, atomicfu_version, "atomicfu")
-        check(version, coroutines_version, "coroutines")
-        check(version, serialization_version, "serialization")
-    }
-    repositories {
-        mavenLocal()
-        maven(url = "https://oss.sonatype.org/content/repositories/snapshots")
+    if (!rootProject.hasProperty("skip_snapshot_checks")) {
```

Comment:
What is this property and where will it be used?

Reply:
Sorry, I see we use it in our config when building libraries

Reply:
But do I understand correctly that this parameter only makes sense when building ktor?

Reply:
for atomicfu, coroutines, serialization in project branches I also see logic with this parameter. 
I will add it to ktor build for train.

Reply:
This property might be useful for running train locally with different versions, but I don't think it should ever be used on CI